### DISCUSSION TITLE
how do I send data from the gap8 to an application on the Crazyflie - objects detection

### DISCUSSION THREAD

**Discussion by crazyflie-aljanabi** [bitcraze/discussions#612] [General]
Hi all,

 I have successfully performed the 'Flash Wifi Example' and I can now see the image from the AI-deck camera on my computer!

I have now another topic: I would like to implement and test object detection for my crazyflie quadcopter on the host computer. The crazyflie must be capable of performing automated flight functions based on camera-based object detection. How can I do this? I need your help please on this topic. What should I do and which code should I use for this?

Best regards
Ali

**Comment by hmllr** [comment]
Hi Ali,
You can use cpx for this - to get started I recommend to you to look at the "Hello World" example (https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/simple-examples/hello-world/) and the blogpost from last Monday which has some explanations about it (https://www.bitcraze.io/2023/03/ai-deck-documentation-and-examples-update/).
Then, on the Crazyflie side you'll need to receive it, there you'll most likely want to use the app layer: https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/userguides/app_layer/
The receiving side infrastructure is unfortunately missing some little parts and the example - but you can get some inspiration in this PR in progress: https://github.com/bitcraze/aideck-gap8-examples/pull/121
Hope this helps to get you started!
Hanna
